internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -460,6 +481,132 @@ func convertEnvSecurityGroupCfg(mft *manifest.Environment) (*template.SecurityGr
 	}, nil
 }
 
+func (s *LoadBalancedWebService) convertApplicationLoadBalancer() (*template.ALBListener, error) {
+	albConfig := s.manifest.RoutingRule
+	if albConfig.Disabled() || albConfig.IsEmpty() {
+		return nil, nil
+	}
+	var rules []template.ALBListenerRule
+	// build listener rule config from primary rule config from manifest.
+	rule, err := RoutingRuleConfigOrBoolConverter{
+		rule:         albConfig,
+		manifest:     s.manifest,
+		httpsEnabled: s.httpsEnabled,
+	}.convert()
+	if err != nil {
+		return nil, nil
+	}
+	aliasesFor, err := convertHostedZone(albConfig.Alias, albConfig.HostedZone)
+	if err != nil {
+		return nil, fmt.Errorf(`convert "http.alias" to string slice: %w`, err)
+	}
+	rules = append(rules, *rule)
+
+	// TODO: @pbhingre build listener rule config from additional rules from manifest.
+
+	httpRedirect := true
+	if albConfig.RedirectToHTTPS != nil {
+		httpRedirect = aws.BoolValue(albConfig.RedirectToHTTPS)
+	}
+
+	config := &template.ALBListener{
+		Rules:             rules,
+		RedirectToHTTPS:   httpRedirect,
+		IsHTTPS:           s.httpsEnabled,
+		HostedZoneAliases: aliasesFor,
+	}
+
+	return config, nil
+}
+
+func (s *BackendService) convertApplicationLoadBalancer() (*template.ALBListener, error) {
+	albConfig := s.manifest.RoutingRule
+	if albConfig.IsEmpty() {
+		return nil, nil
+	}
+	var rules []template.ALBListenerRule
+	// build listener rule config from primary rule config from manifest.
+	rule, err := routingRuleConfigConverter{
+		rule:         albConfig,
+		manifest:     s.manifest,
+		httpsEnabled: s.httpsEnabled,
+	}.convert()
+	if err != nil {
+		return nil, nil
+	}
+	rules = append(rules, *rule)
+	hostedZoneAliases, err := convertHostedZone(albConfig.Alias, albConfig.HostedZone)
+	if err != nil {
+		return nil, fmt.Errorf(`convert "http.alias" to string slice: %w`, err)
+	}
+
+	// TODO: @pbhingre build listener rule config from additional rules from manifest.
+
+	config := &template.ALBListener{
+		Rules:             rules,
+		RedirectToHTTPS:   s.httpsEnabled,
+		IsHTTPS:           s.httpsEnabled,
+		MainContainerPort: s.manifest.MainContainerPort(),
+		HostedZoneAliases: hostedZoneAliases,
+	}
+
+	return config, nil
+}
+
+type loadBalancerTargeter interface {
+	HTTPLoadBalancerTarget() (string, string, error)
+	MainContainerPort() string
+}
+
+type routingRuleConfigConverter struct {
+	rule         manifest.RoutingRuleConfiguration
+	manifest     loadBalancerTargeter
+	httpsEnabled bool
+}
+
+// RoutingRuleConfigOrBoolConverter holds routing rule config for LBWS.
+type RoutingRuleConfigOrBoolConverter struct {
```

Comment:
Why do we even need this...the only convert method is just calling `routingRuleConfigConverter.convert()`

Reply:
`func (conv RoutingRuleConfigOrBoolConverter) convert() (*template.ALBListenerRule, error)` does not use anything unique about `RoutingRuleConfigOrBool`, which doesn't make sense to me. Why do we not use the `OrBool` part at all?

Reply:
if we don't need `OrB ool` logic I still think we can just use `routingRuleConfigConverter` because when we initialize the struct we can use `rule.RoutingRuleConfiguration` (rule is of type `RoutingRuleConfigOrBool`)